phones/models.py

```diff
@@ -201,7 +203,8 @@ def calls_and_texts_blocked(self):
         return self.calls_blocked + self.texts_blocked
 
     def save(self, *args, **kwargs):
-        if not get_verified_realphone_records(self.user):
+        realphone = get_verified_realphone_records(self.user).first()
```

Comment:
Nit:
```suggestion
        realphone = get_verified_realphone_record_qs(self.user).first()
```
This function is way too close to the `get_verified_realphone_record` even though they receive different parameter. My recommendation is to rename the function to better indicate that we are returning a query set or combine this function to existing `get_verified_realphone_record`. However, this is a nice to have refactor.

Reply:
I agree. I think `records` implies a list or queryset, and I was confused about the different parameter type of `record`. I'd go with:

```
get_verified_realphone_records_for_user(User)
get_verified_realphone_record_by_number(number)
```

I also think this is a good opportunity for typing annotations.

However, I'm going to save the refactoring for a future PR, and focus this one on the blocking feature.